File: solutions/feb_2024/1272_remove_interval/solution.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {
  public List<List<Integer>> removeInterval(int[][] intervals, int[] toBeRemoved) {
    List<List<Integer>> result = new ArrayList<>();
    int s1 = toBeRemoved[0];
    int e1 = toBeRemoved[1];

    for (int[] interval : intervals) {
      int s2 = interval[0];
      int e2 = interval[1];

      if (e2 <= s1 || e1 <= s2) {
        result.add(Arrays.asList(s2, e2));
        continue;
      }

      if (s2 < s1) {
        result.add(Arrays.asList(s2, s1));
      }
      if (e1 < e2) {
        result.add(Arrays.asList(e1, e2));
      }
    }

    return result;
  }
}

File: solutions/feb_2024/1272_remove_interval/solution.py
from typing import List


class Solution:
    def removeInterval(
        self, intervals: List[List[int]], toBeRemoved: List[int]
    ) -> List[List[int]]:
        result = []
        s2, e2 = toBeRemoved

        for s1, e1 in intervals:
            if e1 <= s2 or s1 >= e2:
                result.append([s1, e1])
                continue

            if s1 < s2:
                result.append([s1, s2])
            if e2 < e1:
                result.append([e2, e1])

        return result
